Bottom: 4584dc90fb90d84f50a233ca76127ac63112a35c
Top:    857ebb92007f9ece965ac5ddd2dff5ae8273fddf
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-25 01:40:35 +0100

Refresh of hack-play-with-where-the-gdk-v

---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index ea2c783..a3a0321 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -916,6 +916,8 @@ ghid_invalidate_lr (int left, int right, int top, int bottom)
   ghid_invalidate_all ();
 }
 
+void DrawAttached (bool);
+
 void
 ghid_invalidate_all ()
 {
@@ -977,8 +979,10 @@ ghid_invalidate_all ()
 
   hid_expose_callback (&ghid_hid, &region, 0);
   ghid_draw_grid ();
-  if (ghidgui->need_restore_crosshair)
+  if (ghidgui->need_restore_crosshair) {
     RestoreCrosshair (FALSE);
+    DrawAttached (FALSE);
+  }
   ghidgui->need_restore_crosshair = FALSE;
   ghid_screen_update ();
 }
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 0952462..3c7f950 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -228,6 +228,8 @@ ghid_get_coords (const char *msg, int *x, int *y)
     }
 }
 
+void DrawAttached (bool);
+
 gboolean
 ghid_note_event_location (GdkEventButton * ev)
 {
@@ -248,6 +250,7 @@ ghid_note_event_location (GdkEventButton * ev)
   gport->view_x = event_x * gport->zoom + gport->view_x0;
   gport->view_y = event_y * gport->zoom + gport->view_y0;
 
+  DrawAttached (false);
   moved = MoveCrosshairAbsolute (SIDE_X (gport->view_x), 
 				 SIDE_Y (gport->view_y));
   if (moved)
@@ -255,6 +258,7 @@ ghid_note_event_location (GdkEventButton * ev)
       AdjustAttachedObjects ();
       RestoreCrosshair (false);
     }
+  DrawAttached (false);
   ghid_set_cursor_position_labels ();
   return moved;
 }
